test/client.test.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const { test, beforeEach } = require('tap')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+
+class Stub {
+  read() {}
+}
+
+const read = sinon.stub(Stub.prototype, 'read')
+
+const HashiCorpClient = proxyquire('../lib/client', {
+  'node-vault': () => new Stub()
+})
+
+beforeEach(async () => {
+  read.resolves()
+})
+
+test('get', (t) => {
+  t.plan(2)
+
+  t.test('read', async (t) => {
+    t.plan(3)
+
+    const client = new HashiCorpClient({ mountPoint: 'unit-test-secrets' })
+    read.resolves({
+      data: {
+        value: 'secret payload'
+      }
+    })
+
+    const secret = await client.get('name')
+
+    t.ok(read.called, 'calls read')
+    t.ok(read.calledWith('unit-test-secrets/name'), 'provides name to read')
+    t.equal(secret, 'secret payload', 'extracts SecretString')
+  })
+
+  t.test('sdk error', async (t) => {
+    t.plan(1)
+    const client = new HashiCorpClient()
+
+    read.rejects(new Error())
+
+    const promise = client.get('unit-test-secrets/name')
+
+    await t.rejects(promise, 'throws error')
```

Comment:
nit: awaiting a promise that's the last statement of an async function is unnecessary, you can simply return the promise

Reply:
I feel that as this is a test (e.g. it could be expanded on with additional assertions below in the future and the return value of this assertion is not relevant) awaiting the assertion rather than returning it makes the code clearer.

Reply:
yeah I agree that the purpose is clearer this way